src/Cell.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <cassert>
+#include <vector>
+
+#include "Config.h"
+#include "Gas.h"
+#include "Logging.h"
+#include "Parameters.h"
```

Comment:
do we need all these includes in the header file already? If not, I'd rather have them in the cpp file. For example, I'm suspecting we don't really use anything from Parameters.h in this header file.